<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index â€¢ ribiosIO</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ribiosIO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/bedapub/ribiosIO/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="GctMatrix.html">GctMatrix()</a></code> </p>
        </td>
        <td><p>Create a GctMatrix object</p></td>
      </tr><tr><td>
          <p><code><a href="as.matrix.GctMatrix.html">as.matrix(<i>&lt;GctMatrix&gt;</i>)</a></code> </p>
        </td>
        <td><p>Coerce a GctMatrix object into a matrix</p></td>
      </tr><tr><td>
          <p><code><a href="as_numeric_matrix.html">as_numeric_matrix()</a></code> </p>
        </td>
        <td><p>Transform a data.frame to a numeric matrix without characters coereced as factors</p></td>
      </tr><tr><td>
          <p><code><a href="cbindGct.html">cbindGct()</a></code> </p>
        </td>
        <td><p>Column bind (cbind) two GctMatrix objects</p></td>
      </tr><tr><td>
          <p><code><a href="find_ampliseq.html">find_ampliseq()</a></code> <code><a href="find_ampliseq.html">read_ampliseq()</a></code> <code><a href="find_ampliseq.html">find_and_read_ampliseq()</a></code> </p>
        </td>
        <td><p>Find and read-in AmpliSeq files</p></td>
      </tr><tr><td>
          <p><code><a href="gctDesc.html">gctDesc()</a></code> </p>
        </td>
        <td><p>Retrieve feature (row) descriptions from a GctMatrix S3-object</p></td>
      </tr><tr><td>
          <p><code><a href="gctMatrix2longdf.html">gctMatrix2longdf()</a></code> </p>
        </td>
        <td><p>Convert a GctMatrix into a long data frame</p></td>
      </tr><tr><td>
          <p><code><a href="getDataDir.html">getDataDir()</a></code> </p>
        </td>
        <td><p>Get the data directory</p></td>
      </tr><tr><td>
          <p><code><a href="iofile.html">iofile()</a></code> </p>
        </td>
        <td><p>Get file names for data import/export</p></td>
      </tr><tr><td>
          <p><code><a href="isGctFile.html">isGctFile()</a></code> </p>
        </td>
        <td><p>Test a file is a GCT file or not</p></td>
      </tr><tr><td>
          <p><code><a href="is_factor_file.html">is_factor_file()</a></code> <code><a href="is_factor_file.html">is_cls_file()</a></code> </p>
        </td>
        <td><p>Check if a file encodes a factor</p></td>
      </tr><tr><td>
          <p><code><a href="loadFile.html">loadFile()</a></code> </p>
        </td>
        <td><p>Attempt to load a binary RData file</p></td>
      </tr><tr><td>
          <p><code><a href="loadObject.html">loadObject()</a></code> </p>
        </td>
        <td><p>Load an object by its name from a RData file</p></td>
      </tr><tr><td>
          <p><code><a href="loadObjectInEnv.html">loadObjectInEnv()</a></code> </p>
        </td>
        <td><p>Load objects from a RData file and return them in an environment</p></td>
      </tr><tr><td>
          <p><code><a href="loadRDS.html">loadRDS()</a></code> </p>
        </td>
        <td><p>Load an object from a RDS file and returns a logical flag</p></td>
      </tr><tr><td>
          <p><code><a href="longdf2gctMatrix.html">longdf2gctMatrix()</a></code> </p>
        </td>
        <td><p>Convert a long data.frame into a GctMatrix</p></td>
      </tr><tr><td>
          <p><code><a href="optional_suppress_warning.html">optional_suppress_warning()</a></code> </p>
        </td>
        <td><p>Supress warning optionally</p></td>
      </tr><tr><td>
          <p><code><a href="print.GctMatrix.html">print(<i>&lt;GctMatrix&gt;</i>)</a></code> </p>
        </td>
        <td><p>Print method for GctMatrix object</p></td>
      </tr><tr><td>
          <p><code><a href="readMatrix.html">readMatrix()</a></code> </p>
        </td>
        <td><p>Read in numeric matrix from tab-delimited format written by <code>writeMatrix</code></p></td>
      </tr><tr><td>
          <p><code><a href="readTable.html">readTable()</a></code> </p>
        </td>
        <td><p>Read in data.frame from tab-delimited format written by <code>writeMatrix</code></p></td>
      </tr><tr><td>
          <p><code><a href="read_ampliseq_bedcovgct.html">read_ampliseq_bedcovgct()</a></code> </p>
        </td>
        <td><p>Read bedcov output of AmpliSeq amplicons and convert them to read counts</p></td>
      </tr><tr><td>
          <p><code><a href="read_annotated_ampliseq_amplicons.html">read_annotated_ampliseq_amplicons()</a></code> </p>
        </td>
        <td><p>Read AmpliSeq amplicon informaiton from an annotated BED file</p></td>
      </tr><tr><td>
          <p><code><a href="read_bed.html">read_bed()</a></code> </p>
        </td>
        <td><p>Read a BED file</p></td>
      </tr><tr><td>
          <p><code><a href="read_biokit_exprs.html">read_biokit_exprs()</a></code> </p>
        </td>
        <td><p>qRead BioKit expression file into a data.frame</p></td>
      </tr><tr><td>
          <p><code><a href="read_chip.html">read_chip()</a></code> </p>
        </td>
        <td><p>Read CHIP file</p></td>
      </tr><tr><td>
          <p><code><a href="read_david.html">read_david()</a></code> </p>
        </td>
        <td><p>Read tab-delimited result files from DAVID Bioinformatics Resources</p></td>
      </tr><tr><td>
          <p><code><a href="read_exprs_matrix.html">read_exprs_matrix()</a></code> </p>
        </td>
        <td><p>Read an expression matrix from file</p></td>
      </tr><tr><td>
          <p><code><a href="read_factor.html">read_factor()</a></code> <code><a href="read_factor.html">read_cls()</a></code> </p>
        </td>
        <td><p>Read in a factor writtin in the CLS format</p></td>
      </tr><tr><td>
          <p><code><a href="read_fasta.html">read_fasta()</a></code> <code><a href="read_fasta.html">write_fasta()</a></code> </p>
        </td>
        <td><p>Read (write) FASTA sequences into (from) named character vectors</p></td>
      </tr><tr><td>
          <p><code><a href="read_gct_matrix.html">read_gct_matrix()</a></code> <code><a href="read_gct_matrix.html">read_gctstr_matrix()</a></code> </p>
        </td>
        <td><p>Calling C routine to read GCT file into a matrix</p></td>
      </tr><tr><td>
          <p><code><a href="read_gmt_dataframe.html">read_gmt_dataframe()</a></code> </p>
        </td>
        <td><p>Read gene-sets in a GMT file into a data.frame</p></td>
      </tr><tr><td>
          <p><code><a href="read_gmt_list.html">read_gmt_list()</a></code> </p>
        </td>
        <td><p>Calling C routine to read GMT file into a list</p></td>
      </tr><tr><td>
          <p><code><a href="read_illumina_sampleSheet.html">read_illumina_sampleSheet()</a></code> </p>
        </td>
        <td><p>Read the Data block of Illumina sample sheet as data.frame</p></td>
      </tr><tr><td>
          <p><code><a href="read_pheno.html">read_pheno()</a></code> <code><a href="read_pheno.html">read_pheno_factor()</a></code> </p>
        </td>
        <td><p>Read pheno data from CLS or tab-delimited file</p></td>
      </tr><tr><td>
          <p><code><a href="read_raw_ampliseq_amplicons.html">read_raw_ampliseq_amplicons()</a></code> </p>
        </td>
        <td><p>Read AmpliSeq amplicon informaiton from an raw BED file</p></td>
      </tr><tr><td>
          <p><code><a href="read_trimmed_lines.html">read_trimmed_lines()</a></code> </p>
        </td>
        <td><p>Read lines, thereby trimming empty spaces around the strings and removing empty lines</p></td>
      </tr><tr><td>
          <p><code><a href="ribiosIO-package.html">ribiosIO</a></code> <code><a href="ribiosIO-package.html">ribiosIO-package</a></code> </p>
        </td>
        <td><p>High-performance file input/output with ribios</p></td>
      </tr><tr><td>
          <p><code><a href="setDataDir.html">setDataDir()</a></code> </p>
        </td>
        <td><p>Set the data directory</p></td>
      </tr><tr><td>
          <p><code><a href="strList2DataFrame.html">strList2DataFrame()</a></code> </p>
        </td>
        <td><p>Format a string list into a data.frame</p></td>
      </tr><tr><td>
          <p><code><a href="sub-.GctMatrix.html">`[`(<i>&lt;GctMatrix&gt;</i>)</a></code> </p>
        </td>
        <td><p>Subsetting for GctMatrix</p></td>
      </tr><tr><td>
          <p><code><a href="write.tableList.html">write.tableList()</a></code> </p>
        </td>
        <td><p>Write a list of data.frames (tables) into files</p></td>
      </tr><tr><td>
          <p><code><a href="writeMatrix.html">writeMatrix()</a></code> </p>
        </td>
        <td><p>Export matrix into a commonly used tab-delimited format inside Roche Bioinformatics</p></td>
      </tr><tr><td>
          <p><code><a href="writeMatrix.tableList.html">writeMatrix.tableList()</a></code> </p>
        </td>
        <td><p>Write a list of data.frames (tables) into file with writeMatrix</p></td>
      </tr><tr><td>
          <p><code><a href="writeStrList.html">writeStrList()</a></code> </p>
        </td>
        <td><p>Write a list of strings in a tab-delimited file</p></td>
      </tr><tr><td>
          <p><code><a href="write_annotated_ampliseq_amplicons.html">write_annotated_ampliseq_amplicons()</a></code> </p>
        </td>
        <td><p>Write AmpliSeq amplicon informaiton into an annotated BED file</p></td>
      </tr><tr><td>
          <p><code><a href="write_factor.html">write_factor()</a></code> <code><a href="write_factor.html">write_cls()</a></code> </p>
        </td>
        <td><p>Write a factor in the CLS format</p></td>
      </tr><tr><td>
          <p><code><a href="write_gct.html">write_gct()</a></code> </p>
        </td>
        <td><p>Write matrix in GCT file format</p></td>
      </tr><tr><td>
          <p><code><a href="write_gmt.html">write_gmt()</a></code> </p>
        </td>
        <td><p>Write a list of gene sets into a GMT file</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jitao David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

